src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -281,6 +282,49 @@ def with_src_position(
     return expr_ast
 
 
+def _fill_ast_with_column(
```

Comment:
Need a better fn name. One idea would be `set_column_ast`, but you might come up with something better. The param `ast` should probably be called `builder` or sth to indicate that's the value that gets set, otherwise, especially given the current name of the function, the reader gets the opposite idea.